### PR TITLE
fix(database) the core and notary pods cannot start up

### LINKED ISSUE #528: The core and notary pods cannot start up when reinstall the Harbor instance
It seems that the secret which contains the database password isn't deleted when removing the Harbor instance, this causes the core and notary pods reporting invalid database password when starting up

### PR DESCRIPTION
**Pull request by Ghostbaby**
The core and notary pods cannot start up when reinstall the Harbor
instance

fix #528

Signed-off-by: zhuhuijun <zhuhuijunzhj@gmail.com>

### CODE REVIEW

File: pkg/cluster/controllers/database/readiness.go [hunk 1]
```diff
@@ -106,22 +102,27 @@ func getDatabasePasswordRefName(name string) string {
 }
 
 // DeployComponentSecret deploy harbor component database secret.
-func (p *PostgreSQLController) DeployComponentSecret(ctx context.Context, conn *Connect, ns string, secretName string) (*corev1.Secret, error) {
+func (p *PostgreSQLController) DeployComponentSecret(ctx context.Context, conn *Connect, ns string, secretName string,
```
Comment (steven-zou) [inline pkg/cluster/controllers/database/readiness.go:105]: If the whole `*goharborv1.HarborCluster` has been passed into the func arguments, the namespace arg can be removed, right?  @Ghostbaby

File: pkg/cluster/controllers/database/readiness.go [hunk 2]
```diff
@@ -106,22 +102,27 @@ func getDatabasePasswordRefName(name string) string {
...
+	harborcluster *goharborv1.HarborCluster) error {
+
 	secret := &corev1.Secret{}
 	sc := p.GetDatabaseSecret(conn, ns, secretName)
+	if err := controllerutil.SetControllerReference(harborcluster, sc, p.Scheme); err != nil {
```
Comment (steven-zou) [inline pkg/cluster/controllers/database/readiness.go:110]: Why let the cluster own the secret? The cluster has owned the psql CR, so is it possible to let the psql CR own the secret?

**Review by steven-zou — CHANGES_REQUESTED** [review]
Please check comments

File: pkg/cluster/controllers/database/readiness.go (hunk 1 above)
Comment (Ghostbaby) [reply pkg/cluster/controllers/database/readiness.go]: yes.

File: pkg/cluster/controllers/database/readiness.go (hunk 2 above)
Comment (Ghostbaby) [reply pkg/cluster/controllers/database/readiness.go:109]: This secret is store pg conn info for notary and core component, so harborcluster should own this secret.

Comment (steven-zou) [reply pkg/cluster/controllers/database/readiness.go:109]: 👌

**Review by steven-zou — APPROVED** [review]
LGTM

**Review by heww — APPROVED** [review]

**Pull request merged**